cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -132,6 +132,7 @@ func (p *BicepProvider) Plan(
 ) *async.InteractiveTaskWithProgress[*DeploymentPlan, *DeploymentPlanningProgress] {
 	return async.RunInteractiveTaskWithProgress(
 		func(asyncContext *async.InteractiveTaskContextWithProgress[*DeploymentPlan, *DeploymentPlanningProgress]) {
+			p.console.ShowSpinner(ctx, "Creating a deployment plan", input.Step)
```

Comment:
Is there anyway for our providers to not be aware of the complexity of when to start/stop the spinner.  I feel like this is a concern that shouldn't be as tightly coupled to a provider implementation.

Reply:
not if we want the providers to change the spinner status (change the message or stop the spinner to interact with the user).

Even with the current model, providers need to use the asyncContext to require interaction. Which is basically making the provider aware about a change needs to be done at that point. Either using asyncContext or directly making the required change, the provider is aware of the required change.

And since the console was there anyway, and the providers are using the console (by using the a `progressDisplay` that is directly created by the provider to handle - resourceManager, p.console and scope - ), it makes sense to just let the provider consume the console as a general in/out interface